backend/app.js

```diff
@@ -22,16 +51,84 @@ const db = {
   ]
 }
 
+//Allow app to use CORS. Important for transferring JSON files.
 app.use(cors())
 
+//Set appropriate headers for server
+app.use(function(req, res, next){
+    res.setHeader('Access-Control-Allow-Origin', 'http://localhost:4200');
+    res.header("Access-Control-Allow-Headers", "Origin, X-Requested0With, Content-Type, Accept");
+    res.header('Access-Control-Allow-Methods', 'POST, GET, PUT, DELETE, OPTIONS');
+    next();
+});
+
+//Support parsing of application/json type post data
+app.use(bodyParser.json());
+
+app.use(bodyParser.urlencoded({extended: true}));
+
+//Tell mongoose to connect to MongoDB instance and look for ember data.
+//If no database exists, it will create one for us.
+mongoose.connect('mongodb://localhost/emberData');
+
 app.get('/', function (req, res) {
     //res.send('Hello World!')
     res.send('hello')
 })
 
+//New lines???
+app.get('/api/', function(req,res){
+    res.send('working');
+});
+
+//route to get all of the registrations
+app.get('/api/registrations', function(req,res){
+    RegistrationModel.find({}, function(err,docs){
+        if(err){
+            res.send({error:res});
+        }
+        else {
+            //Where registration is name of model we will create in ember
+            //Docs is array of the returned document
+            res.send({registration:docs});
+        }
+    });
+});
+
+//route to post registration
+app.post('/api/registrations', function(req,res){
+    var newRegistration = new RegistrationModel({username: req.body.registration.username,
+    password: req.body.registration.password, email: req.body.registration.email});
+    newRegistration.save(function (err){
+      if(err) return handleError(err);
+    });
+});
+
+//route to get all of the blogposts
+app.get('/api/blogposts', function(req,res){
+    BlogpostModel.find({}, function(err,docs){
+        if(err){
+            res.send({error:res});
+        }
+        else {
+            //Where registration is name of model we will create in ember
+            //Docs is array of the returned document
+            res.send({blogpost:docs});
+        }
+    });
+});
+
+//in response send the id and update the store
+//NEED TO INSERT STUFF TO THE DATABASE
```

Comment:
Does this work now? If so we can remove the all caps comment. If not, happy to convert the comment to a TODO